src/Failure.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Particle\Validator;
+
+class Failure
```

Comment:
Wondering why you called this class `Failure`, `Error` feels more appropriate if you ask me.